### PR TITLE
[httputil] add ReadHeaderTimeout

### LINKED ISSUE #3538: Potential Slowloris Attack because ReadHeaderTimeout is not configured in the http.Server
### Description of the code refactor
2 related issues in security scan result:
https://github.com/iotexproject/iotex-core/security/code-scanning/58
https://github.com/iotexproject/iotex-core/security/code-scanning/59

### Why is this needed
- [] better/less memory usage
- [] reduce running time
- [] clean (easy-to-understand) data structure or algorithm
- [] remove dead code
- [x] better code design or structure (potential security issue)
- [] reduce/decouple dependencies between code modules
- [] increase test coverage
- [] better code readability

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
similar to https://github.com/iotexproject/iotex-core/commit/0d72eb73842e17be6651169b4b8e05b4a2648504, need to fix httputil as well

Fixes #3538

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [x] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/util/httputil/httputil.go [hunk 1]
```diff
@@ -9,20 +9,51 @@ import (
 )
 
 const (
```
Comment (Liuhaai) [inline pkg/util/httputil/httputil.go:11]: add as default conf instead of const

Comment (dustinxie) [reply pkg/util/httputil/httputil.go:11]: updated in 2nd commit

File: api/http.go [hunk 2]
```diff
@@ -31,17 +32,13 @@ func NewHTTPServer(route string, port int, handler http.Handler) *HTTPServer {
...
-			Addr:              ":" + strconv.Itoa(port),
-			WriteTimeout:      30 * time.Second,
-			ReadHeaderTimeout: 10 * time.Second,
-		},
-	}
 	mux := http.NewServeMux()
 	mux.Handle("/"+route, handler)
-	svr.svr.Handler = mux
-	return svr
+
```
Comment (Liuhaai) [inline api/http.go:37]: remove comment on l30

**Review by Liuhaai — APPROVED** [review]

File: pkg/util/httputil/httputil.go [hunk 3]
```diff
@@ -10,19 +10,49 @@ import (
...
+// DefaultServerConfig is the default server config
+var DefaultServerConfig = serverConfig{
+	ReadHeaderTimeout: 5 * time.Second,
+	ReadTimeout:       30 * time.Second,
+	WriteTimeout:      30 * time.Second,
+	IdleTimeout:       120 * time.Second,
+}
+
+// HeaderTimeout sets header timeout
+func HeaderTimeout(h time.Duration) ServerOption {
```
[the remark this answers is not in the document]
Comment (dustinxie) [reply pkg/util/httputil/httputil.go]: fixed in 3rd commit

File: api/http.go (hunk 2 above)
Comment (dustinxie) [reply api/http.go:36]: fixed in 3rd commit

**Review by huof6829 — APPROVED** [review]

**Pull request merged**